raiden/network/channelgraph.py

```diff
@@ -123,16 +128,26 @@ def get_best_routes(
 
     online_nodes = list()
     unknown_nodes = list()
-    for path in available_paths:
-        partner = path[1]
-        channel = channel_graph.partneraddress_channel[partner]
+
+    neighbors_heap = ordered_neighbors(
+        channel_graph.graph,
+        our_address,
+        target_address,
+    )
+
+    for _, partner_address in neighbors_heap:
```

Comment:
Here you are traversing the heap with the heap invariant `heap[k] <= heap[2*k+1] and heap[k] <= heap[2*k+2]`.

Did you maybe want to first [sort it like a list](https://docs.python.org/2/library/heapq.html#basic-examples) by `heappop()` each element to traverse so that you end up iterating from the routes with the smallest `length` to those with the biggest one?